### PR TITLE
Switch to new data API

### PR DESCRIPTION
**Pull request by ovrheat**
I own one of these, that's why I added data API support. I haven't changed anything about output, included raw hex data output (it was called 'Received data') field that was already present in the module.
Should we leave it as it was originally (about showing also raw data) or not? You decide :)

### CODE REVIEW

File: src/devices/generic_temperature_sensor.c [hunk 1]
```diff
@@ -5,16 +5,25 @@
  * it under the terms of the GNU General Public License as published by
  * the Free Software Foundation; either version 2 of the License, or
  * (at your option) any later version.
+ *
+ * Switched to new data API by 'ovrheat' Nik
  */
```
Comment (merbanan) [inline src/devices/generic_temperature_sensor.c]: I am not so found of these type of changes. This info can be found in the git repository log. If you feel that you contributed significantly then add your name as a copyright line.

File: src/devices/generic_temperature_sensor.c [hunk 2]
```diff
@@ -29,21 +38,41 @@ static int generic_temperature_sensor_callback(bitbuffer_t *bitbuffer) {
...
+	snprintf(received_data,sizeof(received_data),"%02x %02x %02x", bb[1][0], bb[1][1], bb[1][2]);
+	
+	
+	//fprintf(stdout, "\nSensor        = Temperature event\n");
+	//fprintf(stdout, "Device        = %d\n", device);
+	//fprintf(stdout, "Battery?      = %02X\n", battery);
+	//fprintf(stdout, "Temp          = %f\n",fTemp);
+	//fprintf(stdout, "Model         = Generic temperature sensor 1\n");
+	//fprintf(stdout, "Received Data = %02x %02x %02x\n", bb[1][0], bb[1][1], bb[1][2]);
+
```
Comment (merbanan) [inline src/devices/generic_temperature_sensor.c]: Remove all these lines.

File: src/devices/generic_temperature_sensor.c [hunk 3]
```diff
@@ -29,21 +38,41 @@ static int generic_temperature_sensor_callback(bitbuffer_t *bitbuffer) {
...
+	//fprintf(stdout, "Received Data = %02x %02x %02x\n", bb[1][0], bb[1][1], bb[1][2]);
+	
+	
+	data = data_make("time", 			"", 			DATA_STRING, 								time_str,
+                     "model", 			"", 			DATA_STRING, 	"Generic temperature sensor 1",
+		             "id",          	"Id",			DATA_FORMAT,	"\t %d",	DATA_INT,		device,
+                     "temperature_C",	"Temperature",	DATA_FORMAT, 	"%.02f C",	DATA_DOUBLE,	fTemp,
+                     "battery",      	"Battery?",		DATA_INT,    								battery,
+					 "rawdata",			"Raw data",		DATA_STRING,								received_data,
+                     NULL);
```
Comment (merbanan) [inline src/devices/generic_temperature_sensor.c]: Reindent these lines so they are symetrical,

File: src/devices/generic_temperature_sensor.c [hunk 4]
```diff
@@ -29,21 +38,41 @@ static int generic_temperature_sensor_callback(bitbuffer_t *bitbuffer) {
...
+	//fprintf(stdout, "Model         = Generic temperature sensor 1\n");
+	//fprintf(stdout, "Received Data = %02x %02x %02x\n", bb[1][0], bb[1][1], bb[1][2]);
+	
+	
+	data = data_make("time", 			"", 			DATA_STRING, 								time_str,
+                     "model", 			"", 			DATA_STRING, 	"Generic temperature sensor 1",
+		             "id",          	"Id",			DATA_FORMAT,	"\t %d",	DATA_INT,		device,
+                     "temperature_C",	"Temperature",	DATA_FORMAT, 	"%.02f C",	DATA_DOUBLE,	fTemp,
+                     "battery",      	"Battery?",		DATA_INT,    								battery,
+					 "rawdata",			"Raw data",		DATA_STRING,								received_data,
```
Comment (merbanan) [inline src/devices/generic_temperature_sensor.c]: Remove the rawdata member.

**Comment by merbanan** [comment]
Fix the comments and the code is ready for merging.

**Comment by ovrheat** [comment]
All done, it should be ok now. I removed also the part which was printing raw output (included the buffer), and also removed my name. No time to mess with copyright laws today :smile:

**Pull request merged**